File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Tutorials/resourcesandlocalization/worldcalc/vb/math/math.vb
 '==========================================================================
'  File:		math.vb
'
'==========================================================================
Imports System
<Assembly: System.Reflection.AssemblyVersion("1.0.0.0")>
<Assembly: System.Reflection.AssemblyKeyFile("WorldCalcKey.snk")>


Namespace Demo.Localize.Math
    
    
    Public Class InvalidFormulaException
        Inherits ApplicationException
    End Class 'InvalidFormulaException
     
    Public Class IntegerMath
        
        Public Function GetResult(arg1 As Int32, op As [Char], arg2 As Int32) As [String]
            Select Case op
                Case "+"c
                        Return [String].Format("{0:###0}", arg1 + arg2)
                Case "-"c
                        Return [String].Format("{0:###0}", arg1 - arg2)
                Case "/"c
                        Return [String].Format("{0:###0}", arg1 \ arg2)
                Case "*"c
                        Return [String].Format("{0:###0}", arg1 * arg2)
                Case Else
                        Throw New InvalidFormulaException()
            End Select
        End Function 'GetResult
    End Class 'IntegerMath 
End Namespace 'Demo.Localize.Math

File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Tutorials/resourcesandlocalization/worldcalc/cs/math/math.cs
//==========================================================================
//  File:		math.cs
//
//==========================================================================
using System;

[assembly: System.Reflection.AssemblyVersion("1.0.0.0")]
[assembly: System.Reflection.AssemblyKeyFile("WorldCalcKey.snk")]

namespace Demo.Localize.Math
{

   public class InvalidFormulaException : ApplicationException { }

   public class IntegerMath
   {
		public String GetResult(Int32 arg1, Char op, Int32 arg2)
		{
			switch(op)
			{
				case '+':
					return String.Format("{0:###0}", arg1 + arg2);
				case '-':
					return String.Format("{0:###0}", arg1 - arg2);
				case '/':
					return String.Format("{0:###0}", arg1 / arg2);
				case '*':
					return String.Format("{0:###0}", arg1 * arg2);
				default:
					throw new InvalidFormulaException();
			}
		}

   }

}